ansible/roles/etesync/tasks/main.yml

```diff
@@ -0,0 +1,203 @@
+  command: find . -name '*.pyc' -delete
+  args:
+    chdir: '{{ etesync__git_checkout }}'
+  become: True
+  become_user: '{{ etesync__user }}'
+  when: etesync__register_source is changed
+
+# Configure EteSync server [[[1
+- name: Exclude secret files from etckeeper
+  copy:
+    content: |
+      secret.txt
+    dest: '{{ etesync__etc + "/.gitignore" }}'
+    owner: 'root'
+    group: 'root'
+    mode: '0600'
+  tags: [ 'role::etesync:config' ]
+
+- name: Generate EteSync configuration
+  template:
+    src: 'etc/etesync-server/etesync-server.ini.j2'
+    dest: '{{ etesync__etc + "/etesync-server.ini" }}'
+    owner: 'root'
+    group: '{{ etesync__group }}'
+    mode: '0640'
+  register: etesync__register_config
+  notify: [ 'Restart gunicorn for etesync' ]
+  tags: [ 'role::etesync:config' ]
+
+- name: Generate extended EteSync configuration
+  template:
+    src: 'usr/local/lib/etesync/app/etesync_site_settings.py.j2'
+    dest: '{{ etesync__git_checkout + "/etesync_site_settings.py" }}'
+    owner: '{{ etesync__user }}'
+    group: '{{ etesync__group }}'
+    mode: '0640'
+  register: etesync__register_config
+  notify: [ 'Restart gunicorn for etesync' ]
+  tags: [ 'role::etesync:config' ]
+
+- name: Generate EteSync secret.txt file
+  template:
+    src: 'etc/etesync-server/secret.txt.j2'
+    dest: '{{ etesync__config_secret_key_filepath }}'
+    owner: 'root'
+    group: '{{ etesync__group }}'
+    mode: '0640'
+  notify: [ 'Restart gunicorn for etesync' ]
+  tags: [ 'role::etesync:config' ]
+
+- name: Perform database installation or migration
+  shell: |
+    ## -o nounset can be enabled with virtualenv >= 16.2.
+    ## Ref: https://github.com/pypa/virtualenv/pull/922
+    set -o pipefail -o errexit;
+    source {{ etesync__virtualenv }}/bin/activate
+    ./manage.py migrate
+  args:
+    chdir: '{{ etesync__git_checkout }}'
+    executable: 'bash'
+  become: True
+  become_user: '{{ etesync__user }}'
+  changed_when: ("No migrations to apply." not in etesync__register_migration.stdout)
+  when: etesync__register_source is changed or etesync__register_config is changed
+  register: etesync__register_migration
+
+- name: Create superuser account
+  environment:
+    SUPERUSER_NAME: '{{ etesync__superuser_name }}'
+    SUPERUSER_EMAIL: '{{ etesync__superuser_email }}'
+    SUPERUSER_PASSWORD: '{{ etesync__superuser_password }}'
+  shell: |
+    ## -o nounset can be enabled with virtualenv >= 16.2.
+    ## Ref: https://github.com/pypa/virtualenv/pull/922
+    set -o pipefail -o errexit;
+    source {{ etesync__virtualenv }}/bin/activate
+    echo "from django.contrib.auth.models import User;
+          User.objects.create_superuser('${SUPERUSER_NAME}', '${SUPERUSER_EMAIL}', '${SUPERUSER_PASSWORD}')" | ./manage.py shell
+  args:
+    chdir: '{{ etesync__git_checkout }}'
+    executable: 'bash'
+  become: True
+  become_user: '{{ etesync__user }}'
+  when: ("etesync" not in ansible_local)
+  no_log: True
```

Comment:
After the change the task is executed, but (on Debian Buster) it errors out with:
```
fatal: [etesync]: FAILED! => {"changed": true, "cmd": "## -o nounset can be enabled with virtualenv >= 16.2.\n## Ref: https://github.com/pypa/virtualenv/pull/922\nset -o pipefail -o errexit;\nsource /usr/local/lib/etesync/virtualenv/bin/activate\necho \"from django.contrib.auth.models import User;\n      User.objects.create_superuser('${SUPERUSER_NAME}', '${SUPERUSER_EMAIL}', '${SUPERUSER_PASSWORD}')\" | ./manage.py shell\n", "delta": "0:00:00.182531", "end": "2020-01-29 22:54:04.625757", "msg": "non-zero return code", "rc": 1, "start": "2020-01-29 22:54:04.443226", "stderr": "Traceback (most recent call last):\n  File \"./manage.py\", line 22, in <module>\n    execute_from_command_line(sys.argv)\n  File \"/usr/local/lib/etesync/virtualenv/lib/python3.7/site-packages/django/core/management/__init__.py\", line 381, in execute_from_command_line\n    utility.execute()\n  File \"/usr/local/lib/etesync/virtualenv/lib/python3.7/site-packages/django/core/management/__init__.py\", line 375, in execute\n    self.fetch_command(subcommand).run_from_argv(self.argv)\n  File \"/usr/local/lib/etesync/virtualenv/lib/python3.7/site-packages/django/core/management/base.py\", line 316, in run_from_argv\n    self.execute(*args, **cmd_options)\n  File \"/usr/local/lib/etesync/virtualenv/lib/python3.7/site-packages/django/core/management/base.py\", line 353, in execute\n    output = self.handle(*args, **options)\n  File \"/usr/local/lib/etesync/virtualenv/lib/python3.7/site-packages/django/core/management/commands/shell.py\", line 92, in handle\n    exec(sys.stdin.read())\n  File \"<string>\", line 2\n    User.objects.create_superuser('drybjed', 'drybjed', 'fE0hheO78pC16Ths.U2Z')\n    ^\nIndentationError: unexpected indent", "stderr_lines": ["Traceback (most recent call last):", "  File \"./manage.py\", line 22, in <module>", "    execute_from_command_line(sys.argv)", "  File \"/usr/local/lib/etesync/virtualenv/lib/python3.7/site-packages/django/core/management/__init__.py\", line 381, in execute_from_command_line", "    utility.execute()", "  File \"/usr/local/lib/etesync/virtualenv/lib/python3.7/site-packages/django/core/management/__init__.py\", line 375, in execute", "    self.fetch_command(subcommand).run_from_argv(self.argv)", "  File \"/usr/local/lib/etesync/virtualenv/lib/python3.7/site-packages/django/core/management/base.py\", line 316, in run_from_argv", "    self.execute(*args, **cmd_options)", "  File \"/usr/local/lib/etesync/virtualenv/lib/python3.7/site-packages/django/core/management/base.py\", line 353, in execute", "    output = self.handle(*args, **options)", "  File \"/usr/local/lib/etesync/virtualenv/lib/python3.7/site-packages/django/core/management/commands/shell.py\", line 92, in handle", "    exec(sys.stdin.read())", "  File \"<string>\", line 2", "    User.objects.create_superuser('drybjed', 'drybjed', 'fE0hheO78pC16Ths.U2Z')", "    ^", "IndentationError: unexpected indent"], "stdout": "", "stdout_lines": []}
```
I think that making it closer to the similar task in the `netbox` role by changing the content to not use the `: |` syntax will fix this, but it's getting late here so I'm heading to bed. I'll try and fix it tommorow if @ypid won't do it first. :-)

Reply:
Thanks for the review and the hint. Fixed it.